src/ports/rt-kernel/pnal_snmp.c

```diff
@@ -0,0 +1,103 @@
+/*********************************************************************
+ *        _       _         _
+ *  _ __ | |_  _ | |  __ _ | |__   ___
+ * | '__|| __|(_)| | / _` || '_ \ / __|
+ * | |   | |_  _ | || (_| || |_) |\__ \
+ * |_|    \__|(_)|_| \__,_||_.__/ |___/
+ *
+ * www.rt-labs.com
+ * Copyright 2020 rt-labs AB, Sweden.
+ *
+ * This software is dual-licensed under GPLv3 and a commercial
+ * license. See the file LICENSE.md distributed with this software for
+ * full license information.
+ ********************************************************************/
+
+#include "options.h"
+#include "pnet_api.h"
+#include "osal.h"
+#include "osal_log.h"
+#include "pnal.h"
+#include "pf_types.h"
+#include "pf_snmp.h"
+#include "pnal_snmp.h"
+#include "lldp-mib.h"
+#include <lwip/apps/snmp.h>
+#include <lwip/apps/snmp_mib2.h>
+#include <lwip/apps/snmp_threadsync.h>
+
+/* Global variable for SNMP server state */
+pnal_snmp_t pnal_snmp;
+
+/**
+ * Called right after writing a variable
+ *
```

Comment:
I guess the function prototype for this type of callback is defined somewhere in lwip-snmp. Please mention the type/name of that.
Probably:;
`This is a callback for snmp in lwip. Arguments should fulfill snmp_write_callback_fct`

Reply:
Done.